### ISSUE TITLE
Does not work with node version 0.12 and newer

### ISSUE THREAD

**Issue by vibgy** [plus3network/fit#6]
Because V8 changed API, this npm does not work with newer version. The easy fix is to update the java npm version. Newer version of java npm can work with newer node versions. Perhaps some package.json magic to enable both.

**Comment by contra** [comment]
PR to update java in the package.json?

**Comment by vibgy** [comment]
Using the npm that comes with node 0.10.0 I was not able to install node-gyp

```
npm ERR! Error: No compatible version found: graceful-fs@'^4.1.2'
npm ERR! Valid install targets:
npm ERR! ["1.0.0","1.0.1","1.0.2","1.1.0","1.1.1","1.1.2","1.1.3","1.1.4","1.1.5","1.1.6","1.1.7","1.1.8","1.1.9","1.1.10","1.1.11","1.1.12","1.1.13","1.1.14","1.2.0","1.2.1","1.2.2","1.2.3","2.0.0","2.0.1","2.0.2","2.0.3","3.0.0","3.0.1","3.0.2","3.0.3","3.0.4","3.0.5","3.0.6","3.0.7","3.0.8","4.1.0","4.1.1","4.1.2"]
npm ERR!     at installTargetsError 
```

Then I tried updating npm that comes with node 0.10.0, I got the following, still error.
```
npm ERR! tar pack Error reading /var/folders/34/zbfg85w549jcxg8gd4lmct080000gr/T/npm-81957/1443568945627-0.9476439275313169/package
npm ERR! TypeError: Cannot call method 'filter' of undefined
npm ERR!     at Packer.IgnoreReader.addIgnoreRules (/Users/gnolkha/.nvm/v0.10.0/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/ignore.js:148:13)
npm ERR!     at Packer.<anonymous> (/Users/gnolkha/.nvm/v0.10.0/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/ignore.js:133:10)
npm ERR!     at fs.js:252:14
npm ERR!     at Object.oncomplete (fs.js:93:15)
npm ERR! If you need help, you may report this log at:
npm ERR!     <http://github.com/isaacs/npm/issues>
npm ERR! or email it to:
npm ERR!     <npm-@googlegroups.com>
```

But finally I was able to do it using the node version 0.10.40 and the tests pass after that (as long as the hardcoded timezone in the fixtures file is fixed to the person's timezone).

I'll create a PR with node engine > 0.10.40, if thats okay.. And I think its okay because java npm only supports node > 0.10.0 anyway. 

```
// https://github.com/joeferner/node-java/blob/master/package.json
"node": ">=0.10.0"
```

Any comments?

**Comment by contra** [comment]
Published as 0.0.3
